test/assignment-5/CollateralizedVault.t.sol

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import {CollateralizedVault} from "src/assignment-5/CollateralizedVault.sol";
+import {Dai} from "src/assignment-5/DAI.sol";
+import {WETH9} from "src/assignment-5/WETH.sol";
+import {IERC20} from "yield-utils-v2/token/IERC20.sol";
+import {ChainlinkPriceFeedMock} from "src/assignment-5/ChainlinkPriceFeedMock.sol";
+
+abstract contract ZeroState is Test {
+
+    using stdStorage for StdStorage;
+
+    address USER = address(1);
+    CollateralizedVault vault;
+    Dai dai;
+    WETH9 weth;
+    ChainlinkPriceFeedMock priceFeedMock;
+
+    function setUp() public virtual {
+        dai = new Dai(block.chainid);
```

Comment:
i like to add `vm.label(USER, 'user');` here.  it helps clean up the stack trace